pyvista/plotting/renderer.py

```diff
@@ -437,48 +439,49 @@ def disable_depth_peeling(self):
         self.SetUseDepthPeeling(False)
         self.Modified()
 
-    def enable_anti_aliasing(self):
-        """Enable anti-aliasing using FXAA.
+    def enable_anti_aliasing(self, aa_type):
+        """Enable anti-aliasing.
 
-        This tends to make edges appear softer and less pixelated.
+        Parameters
+        ----------
+        aa_type : str
+            Anti-aliasing type. Either ``"fxaa"``, ``"mxaa"``, or ``"ssaa"``.
 
-        Warnings
-        --------
-        Enabling this causes screenshots with vtk compiled with OSMesa to be
-        all black. This cannot be enabled when compiled with OSMesa (EGL). See
-        https://github.com/pyvista/pyvista/issues/2686 for more details.
+        """
+        if not isinstance(aa_type, str):
+            raise TypeError(f'`aa_type` must be a string, not {type(aa_type)}')
+        aa_type = aa_type.lower()
+
+        if aa_type == 'fxaa':
+            if uses_egl():  # pragma: no cover
+                # only display the warning when not building documentation
+                if not pyvista.BUILDING_GALLERY:
+                    warnings.warn(
+                        "VTK compiled with OSMesa does not properly support "
+                        "FXAA anti-aliasing and SXAA will be used instead."
+                    )
+                self._render_passes.enable_ssaa_pass()
+            self._enable_fxaa()
 
-        Examples
-        --------
-        >>> import pyvista
-        >>> pl = pyvista.Plotter()
-        >>> pl.enable_anti_aliasing()
-        >>> _ = pl.add_mesh(pyvista.Sphere(), show_edges=True)
-        >>> pl.show()
+        elif aa_type == 'ssaa':
+            self._render_passes.enable_ssaa_pass()
 
-        """
-        if uses_egl():  # pragma: no cover
-            # only display the warning when not building documentation
-            if not pyvista.BUILDING_GALLERY:
-                warnings.warn(
-                    "VTK compiled with OSMesa does not properly support anti-aliasing and anti-aliasing will not be enabled."
-                )
-            return
-        self.SetUseFXAA(True)
-        self.Modified()
+        else:
+            raise ValueError(f'Invalid `aa_type` "{aa_type}". Should be either "fxaa" or "ssaa"')
 
     def disable_anti_aliasing(self):
-        """Disable anti-aliasing.
+        """Disable all anti-aliasing."""
+        self._render_passes.disable_ssaa_pass()
+        self.SetUseFXAA(False)
+        self.Modified()
 
-        Examples
-        --------
-        >>> import pyvista
-        >>> pl = pyvista.Plotter()
-        >>> pl.disable_anti_aliasing()
-        >>> _ = pl.add_mesh(pyvista.Sphere(), show_edges=True)
-        >>> pl.show()
```

Comment:
I don't think it's necessary, but this can be up for debate. Typical use is to have the user interact at the `Plotter` level. This example above even directs the user to `disable_anti_aliasing` from the `Plotter` class.